faunadb-java/src/main/java/com/faunadb/client/types/SetRef.java

```diff
@@ -0,0 +1,46 @@
+package com.faunadb.client.types;
+
+import com.fasterxml.jackson.annotation.JsonValue;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.google.common.base.Optional;
+import com.google.common.collect.ImmutableMap;
+
+/**
+ * A FaunaDB set literal.
+ *
+ * @see <a href="https://faunadb.com/documentation/queries#values-special_types">FaunaDB Special Types</a>
+ */
+@JsonDeserialize(using = Codec.SetRefDeserializer.class)
+public class SetRef extends Value.ScalarValue<ImmutableMap<String, Value>> {
+
+  public SetRef(ImmutableMap<String, Value> parameters) {
+    super(parameters);
+  }
+
+  @JsonValue
+  @Override
+  public ImmutableMap<String, Value> asObject() {
+    return value;
+  }
+
+  @Override
+  public Value get(String key) {
```

Comment:
I'm not as sure about this, but I think the structure of SetRefs should not available via the generic Value interface, since they are not containers like Object or Array. Reflecting on the structure should be possible, but it should be differentiated from the generic path traversal APIs.